catalog/dags/elasticsearch_cluster/healthcheck_dag.py

```diff
@@ -0,0 +1,158 @@
+
+EXPECTED_NODE_COUNT = 6
+EXPECTED_DATA_NODE_COUNT = 3
+EXPECTED_MASTER_NODE_COUNT = 3
+
+
+def _format_response_body(response_body: dict) -> str:
+    return f"""
+    Full healthcheck response body:
+    ```
+    {json.dumps(response_body, indent=4)}
+    ```
+    """
+
+
+def _compose_red_status(env: Environment, response_body: dict):
+    message = f"""
+    Elasticsearch {env} cluster status is **red**.
+
+    This is a critical status change, **investigate ASAP**.
+
+    {_format_response_body(response_body)}
+    """
+    return message
+
+
+def _compose_unexpected_node_count(env: Environment, response_body: dict):
+    node_count = response_body["number_of_nodes"]
+    data_node_count = response_body["number_of_data_nodes"]
+    master_node_count = node_count - data_node_count
+
+    message = f"""
+    Elasticsearch {env} cluster node count is **{node_count}**.
+    Expected {EXPECTED_NODE_COUNT} total nodes.
+
+    Master nodes: **{master_node_count}** of expected {EXPECTED_MASTER_NODE_COUNT}
+    Data nodes: **{data_node_count}** of expected {EXPECTED_DATA_NODE_COUNT}
+
+    This is a critical status change, **investigate ASAP**.
+    If this is expected (e.g., during controlled node or cluster changes), acknowledge immediately with explanation.
+
+    {_format_response_body(response_body)}
+    """
+    logger.error(f"Unexpected node count; {json.dumps(response_body)}")
+    return message
+
+
+def _compose_yellow_cluster_health(env: Environment, response_body: dict):
+    message = f"""
+    Elasticsearch {env} cluster health is **yellow**.
+
+    This does not mean something is necessarily wrong, but if this is not expected (e.g., data refresh) then investigate cluster health now.
+
+    {_format_response_body(response_body)}
+    """
+    logger.info(f"Cluster health was yellow; {json.dumps(response_body)}")
+    return message
+
+
+@task
+def ping_healthcheck(env: str, es_host: str):
+    es_conn: Elasticsearch = ElasticsearchPythonHook(hosts=[es_host]).get_conn
+
+    response = es_conn.cluster.health()
+
+    return response.body
+
+
+@task
+def compose_notification(env: Environment, response_body: dict):
+    status = response_body["status"]
+
+    if status == "red":
+        return "alert", _compose_red_status(env, response_body)
+
+    if response_body["number_of_nodes"] != EXPECTED_NODE_COUNT:
+        return "alert", _compose_unexpected_node_count(env, response_body)
+
+    if status == "yellow":
+        return "notification", _compose_yellow_cluster_health(env, response_body)
+
+    logger.info(f"Cluster health was green; {json.dumps(response_body)}")
+    return None, None
+
+
+@task
+def notify(env: str, message_type_and_string: tuple[str, str]):
+    message_type, message = message_type_and_string
+
+    if message_type == "alert":
+        send_alert(message, dag_id=_DAG_ID.format(env=env))
+    elif message_type == "notification":
+        send_message(message, dag_id=_DAG_ID.format(env=env))
```

Comment:
I know this is a simple DAG - maybe we could raise an `AirflowSkipException` in an `else` clause here, or even a `ValueError` if you feel an error is warranted when the `message_type` is unexpected?

Reply:
Done in https://github.com/WordPress/openverse/pull/3748/commits/9ff7608ef3227452e70ebf2c09b34ac5dd0c17c8